jsk_topic_tools/src/jsk_topic_tools/transport.py

```diff
@@ -130,3 +134,47 @@ def advertise(self, *args, **kwargs):
         pub = rospy.Publisher(*args, **kwargs)
         self._publishers.append(pub)
         return pub
+
+
+class DiagnosticTransport(ConnectionBasedTransport):
```

Comment:
instead of create new class, how about update `ConnectionBasedTransport` and enable this feature when  `/use_vital_check_on_connection_based_transport`(or other good variable name) is `true`

Reply:
That's a good idea!
I merge diagnostic function to `ConnectionBasedTransport`.
https://github.com/jsk-ros-pkg/jsk_common/pull/1711/commits/c8efdb1c60f4720b940f7161b1b43b44a4bb3fab